src/tape/ensemble.py

```diff
@@ -1237,6 +1240,165 @@ def _standardize_batch(self, batch, on, by_band):
 
         return batch
 
+    def save_ensemble(self, path=".", dirname="ensemble", additional_frames=True, **kwargs):
+        """Save the current ensemble frames to disk.
+
+        Parameters
+        ----------
+        path: 'str' or path-like, optional
+            A path to the desired location of the top-level save directory, by
+            default this is the current working directory.
+        dirname: 'str', optional
+            The name of the saved ensemble directory, "ensemble" by default.
+        additional_frames: bool, or list, optional
+            Controls whether EnsembleFrames beyond the Object and Source Frames
+            are saved to disk. If True or False, this specifies whether all or
+            none of the additional frames are saved. Alternatively, a list of
+            EnsembleFrame names may be provided to specify which frames should
+            be saved. Object and Source will always be added and do not need to
+            be specified in the list. By default, all frames will be saved.
+        **kwargs:
+            Additional kwargs passed along to EnsembleFrame.to_parquet()
+
+        Returns
+        ----------
+        None
+
+        Note
+        ----
+        If the object frame has no columns, which is often the case when an
+        Ensemble is constructed using only source files/dictionaries, then an
+        object subdirectory will not be created. `Ensemble.from_ensemble` will
+        know how to work with the directory whether or not the object
+        subdirectory is present.
+
+        Be careful about repeated saves to the same directory name. This will
```

Comment:
Given how liberally jupyter users may hit "run all cells", I think this could lead to some easy mistakes.

What about having an "overwrite" parameter?

If True, we will first try to delete the save directory (possibly printing a message if we did so)

If False, we can throw an exception if the save directory already exists noting they should use a different name or set "overwrite=True"

My instinct is that the default value of the parameter should be False.

Reply:
Yeah, so we have access to overwrite already via the **kwargs passed along to all the to_parquet calls. What remains is that that will only overwrite any subdirectories (ensemble/object, ensemble/source, ensemble/result, etc.) that were present in both saves. So any previous subdirectories will not be deleted if a new save doesn't include them. I'm really worried about doing any form of delete, just as someone could specify a non-empty directory and I wouldn't want to delete anything else. 

So for 1. Maybe the overwrite parameter should graduate out of the **kwargs for visibility? And 2. I'm not sure but maybe we could have save write some kind of subdirs metadata file that lets TAPE know which subdirectories it made, so that overwrite can clean those directories up in subsequent overwrite-enabled saves?

Reply:
Wilson and I talked offline a bit on this. The strategy that we agreed on is to generate a metadata file that knows which subdirectories were produced by the save command. Successive save commands will look for this file and use it to locate which subdirectories in the directory should be removed during the save operation. In particular, we'd be looking for any subdirectories created by a previous save command that will not be overwritten by the current save command. Parquet will handle overwrites for conflicting directories, but we will need ensemble.save_ensemble to catch and remove these frame subdirectories that would otherwise not be touched.

Reply:
The latest commit implements this, with the tweak that I ended up opting to just clean all the previous subdirectories up first as it was logically easier.